csharp-basics/exercises/Scooters/Scooters.Test/RentalCompanyTest.cs

```diff
@@ -0,0 +1,215 @@
+        {
+            //Act
+            _scooterService.AddScooter("Mazda03", 0.015m);
+            _scooterService.AddScooter("Mazda04", 0.015m);          
+            _companyA.StartRent("Mazda03");
+
+            //Assert
+            Assert.Throws<ScooterRentedException>(() => _companyA.StartRent("Mazda03"), "No Exception thrown if Scooter already under rent is being rented!");
+            Assert.Throws<ScooterRentedException>(() => _companyA.EndRent("Mazda04"), "No Exception thrown if Scooter not rented is being returned from rental!");
+        }
+
+        [Test]
+        public void IRentalCompany_05StartEndRentOfNonExistingScooter_ExceptionThrown()
+        {
+            //Assert
+            Assert.Throws<ScooterIdNotFoundException>(() => _companyA.StartRent("Mazda003"), "No Exception thrown if Scooter id not found is being rented!");
+            Assert.Throws<ScooterIdNotFoundException>(() => _companyA.EndRent("Mazda004"), "No Exception thrown if Scooter id not found is being returned from rental!");
+        }
+       
+        [Test]
+        public void IRentalCompany_06CalculateIncome2020IfLessThan20EurPer24h_EqualTo18_9()
+        {
+            //Arrange
+            _expectedResult = 18.9m;
+
+            //Act
+            _scooterService.AddScooter("Mazda05", 0.015m);
+            _rentalHistory.Add(new RentalData("Mazda05", new System.DateTime(2020, 7, 12, 10, 10, 0), new System.DateTime(2020, 7, 13, 7, 10, 0), _scooterService.GetScooterById("Mazda05").PricePerMinute));
+           
+            //Assert
+            Assert.AreEqual(_expectedResult, _companyA.CalculateIncome(2020, false), "Income is not calculated correctly");
+        }
+
+        [Test]
+        public void IRentalCompany_07CalculateIncomeCheckYearValidity2030_ExceptionThrown()
+        {
+            //Assert
+            Assert.Throws<InvalidDateTimeException>(() => _companyA.CalculateIncome(2030, true), "No Exception thrown if Income for not allowed period");
+        }
+
+        [Test]
+        public void RentalData_08AddYear1998andStartAfterEnd_ExceptionThrown()
+        {
+            //Act
+            _scooterService.AddScooter("Mazda06", 0.015m);
+
+            //Assert
+            Assert.Throws<InvalidDateTimeException>(() => _rentalHistory.Add(new RentalData("Mazda06", new DateTime(1998, 6, 1), new DateTime(2033, 6, 3), _scooterService.GetScooterById("Mazda06").PricePerMinute)), "No Exception thrown if Years <2010 or current year!");
+            Assert.Throws<InvalidDateTimeException>(() => _rentalHistory.Add(new RentalData("Mazda06", new DateTime(2020, 6, 1), new DateTime(2020, 5, 3), _scooterService.GetScooterById("Mazda06").PricePerMinute)), "No Exception thrown if rental end before start!");
+        }
+
+        [Test]
+        public void IRentalCompany_09CalculateIncomeOfNotActiveScooters_EqualTo47_7()
+        {
+            //Arrange
+            _expectedResult = 47.7m;
+
+            //Act
```

Comment:
Vienā testā viens act ko tad arī pārbauda. Ja šo visu vajag vienam testam, tad to dara pie arrange